Deprecated pulse_lib and renamed it to library

hykavitha opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

#4168 

### Details and comments

        copied pulse_lib and named the copy to  library 
       Added the whole folder
         -> qiskit/pulse/library/

In all these .py files, where ever pulse_lib is called, its replaced by library

        modified:   qiskit/assembler/assemble_schedules.py
	modified:   qiskit/pulse/__init__.py
	modified:   qiskit/pulse/commands/parametric_pulses.py
	modified:   qiskit/pulse/commands/pulse_decorators.py
	modified:   qiskit/pulse/commands/sample_pulse.py
	modified:   qiskit/pulse/instructions/play.py
	modified:   qiskit/pulse/pulse_lib/__init__.py
	modified:   qiskit/qobj/converters/pulse_instruction.py
	modified:   test/python/compiler/test_assembler.py
	modified:   test/python/pulse/test_discrete_pulses.py
	modified:   test/python/pulse/test_instruction_schedule_map.py
	modified:   test/python/pulse/test_instructions.py
	modified:   test/python/pulse/test_pulse_lib.py
	modified:   test/python/pulse/test_samplers.py
	modified:   test/python/pulse/test_schedule.py

totally
 24 files changed

hykavitha commented:
@taalexander @lcapelluto : fixed this issue. test is all passing. please check if anything else needs to be done from my end?

lcapelluto commented:
This is close! We do want to maintain the git blame, though. To do this, you should be able to follow these steps (we can verify that it's worked when you push):
1) Delete all the files in `qiskit/pulse/library/`
2) Move the files from `qiskit/pulse/pulse_lib/` to `qiskit/pulse/library`. When you run `git status`, you should see these files listed as "renamed" and not "new files". Be careful not to make any changes to the files.
3) Commit that change.
4) Make new files in `qiskit/pulse/pulse_lib` similar to how you made them in `library`, but the only contents of each file should be the header, deprecation warnings, and import statements from `library`, so that anyone could get the same objects from the old paths for now. You can use `# pylint: disable=unused-import` to ignore pylint errors that the imported values aren't used

hykavitha commented:
@taalexander & @lcapelluto : I have resolved the conflicts in this branch, but its still showing conficts please check and let me know.  `hykavitha:fixissue_pulse_lib_4168`

taalexander commented:
I am seeing these conflicts. Are you certain you have the most recent version of master when you performed a merge? Or that master wasn't updated while you were resolving conflicts?

hykavitha commented:
> I am seeing these conflicts. Are you certain you have the most recent version of master when you performed a merge? Or that master wasn't updated while you were resolving conflicts?

Hi @taalexander  with the steps given by @lcapelluto I have resolved conflicts & merged my branch fixissue_pulse_lib_4168 with master. please check it and let me know

hykavitha commented:
@taalexander & @lcapelluto : can you please check this pull request? I need to move faster too working on Qiskit Advocacy

lcapelluto commented:
Hi @hykavitha! I'm glad I was able to help you resolve your merge conflicts. Sorry for the delay in a re-review. A failing build signals that something is still in progress -- in this case, please run `make lint; make style` and `python -m unittest` and fix the errors. I'll also update your branch with the latest changes from `master`.

lcapelluto commented on qiskit/pulse/commands/parametric_pulses.py:
```diff
@@ -17,8 +17,8 @@
 
-from qiskit.pulse.pulse_lib import (ParametricPulse, Gaussian, GaussianSquare,
-                                    Drag, Constant, ConstantPulse)
+from qiskit.pulse.library import ParametricPulse, Gaussian, GaussianSquare, Drag, Constant, ConstantPulse
+
```
```suggestion
from qiskit.pulse.pulse_lib import (ParametricPulse, Gaussian, GaussianSquare,
                                    Drag, Constant, ConstantPulse)
```
This is one of the lines causing a style error, because the line length is greater than 100. Since this file is deprecated anyway (all of `commands/` is going to be removed), I don't think we have to worry about updating the import path.

lcapelluto commented:
@hykavitha I've just checked out your changes again. I had made a comment previously, and I don't think you've incorporated this yet:

> This is close! We do want to maintain the git blame, though. To do this, you should be able to follow these steps (we can verify that it's worked when you push):
> 
> 1. Delete all the files in `qiskit/pulse/library/`
> 2. Move the files from `qiskit/pulse/pulse_lib/` to `qiskit/pulse/library`. When you run `git status`, you should see these files listed as "renamed" and not "new files". Be careful not to make any changes to the files.
> 3. Commit that change.
> 4. Make new files in `qiskit/pulse/pulse_lib` similar to how you made them in `library`, but the only contents of each file should be the header, deprecation warnings, and import statements from `library`, so that anyone could get the same objects from the old paths for now. You can use `# pylint: disable=unused-import` to ignore pylint errors that the imported values aren't used
> 
> Great job with this! Thank you again for your contributions!


Are these instructions clear? Git blame allows us to track who wrote what and when. It helps us maintain the code going forward.

hykavitha commented:
> @hykavitha I've just checked out your changes again. I had made a comment previously, and I don't think you've incorporated this yet:
> 
> > This is close! We do want to maintain the git blame, though. To do this, you should be able to follow these steps (we can verify that it's worked when you push):
> > 
> > 1. Delete all the files in `qiskit/pulse/library/`
> > 2. Move the files from `qiskit/pulse/pulse_lib/` to `qiskit/pulse/library`. When you run `git status`, you should see these files listed as "renamed" and not "new files". Be careful not to make any changes to the files.
> > 3. Commit that change.
> > 4. Make new files in `qiskit/pulse/pulse_lib` similar to how you made them in `library`, but the only contents of each file should be the header, deprecation warnings, and import statements from `library`, so that anyone could get the same objects from the old paths for now. You can use `# pylint: disable=unused-import` to ignore pylint errors that the imported values aren't used
> > 
> > Great job with this! Thank you again for your contributions!
> 
> Are these instructions clear? Git blame allows us to track who wrote what and when. It helps us maintain the code going forward.

1. I removed all files in library
2. moved cp -r recursively all files in pulse_lib to library
3. git status -> dint say renamed, it said deleted all files in library & modified pulse_lib files to library
4. created file & folder structure in pulse_lib to just contain header imports like this
import warnings, looks like it is not correct way... its throwing  import errors. please advice me how to make the import work. (I have not pushed this code)

```
import warnings

from qiskit.pulse.library.continuous import *
warnings.warn("the pulse_lib module is deprecated, pulse_lib is renamed to library",
              DeprecationWarning, stacklevel=2)
```

```
from qiskit.pulse.library.samplers.decorators import left, right, midpoint
warnings.warn("the pulse_lib module is deprecated, pulse_lib is renamed to library",
              DeprecationWarning, stacklevel=2)
```

lcapelluto commented:
@hykavitha Hey! I'm ready to merge this as soon as you can fix the merge conflicts. I tried to do it myself, but wasn't able to

lcapelluto commented:
Fixed up the final pieces in #4662
